utilities/bulk-string-replace.py

```diff
@@ -0,0 +1,179 @@
+            (prefix.node or '')[:min([ len(prefix.node or ''), 20 ])],
+            (prefix.description or '')[:min([ len(prefix.description or ''), 900 ])]
+        )
+        if prefix.match:
+            new_prefix_node = re.sub(pattern, replacement, (prefix.node or ''), flags=re.IGNORECASE)
+            new_prefix_desc = re.sub(pattern, replacement, (prefix.description or ''), flags=re.IGNORECASE)
+            print ("\x1b[92m" + print_pattern) % (
+                "+",
+                prefix.vrf.rt,
+                prefix.type[0].upper(),
+                ("  " * prefix.indent + prefix.display_prefix)[:min([ len(prefix.display_prefix) + 2*prefix.indent, 30 ])],
+                new_prefix_node[:min([ len(new_prefix_node), 20 ])],
+                new_prefix_desc[:min([ len(new_prefix_desc), 90 ])]
+            )
+
+
+    # reset colors
+    print "\x1b[0m",
+
+    # Perform action?
+    print "Select replacements to perform"
+    print "Enter comma-separated selection (eg. 5,7,10) or \"all\" for all prefixes."
+    print "Prefix list with ! to invert selection (eg !5,7,10 to perform operation on all except the entered prefixes)"
+    inp = raw_input("Selection: ").strip()
+
+    invert = False
+    if inp[0] == "!":
+        inp = inp[1:]
+        invert = True
+
+    rename_all = False
+    if inp == 'all':
+        rename_all = True
+        selection = []
+    else:
+        selection = inp.split(",")
+        selection = map(lambda x: int(x.strip()), selection)
+
+    for i, prefix in enumerate(prefix_list):
+
+        if prefix.match and ((invert and i not in selection) or (not invert and i in selection) or rename_all):
+            if prefix.node is not None:
+                prefix.node = re.sub(pattern, replacement, prefix.node, flags=re.IGNORECASE)
+            if prefix.description is not None:
+                prefix.description = re.sub(pattern, replacement, prefix.description, flags=re.IGNORECASE)
+
+            print "Saving prefix %s..." % prefix.display_prefix
+            prefix.save()
+
+if __name__ == '__main__':
+
+    # Read config from ~/.nipaprc
+    cfg = ConfigParser.ConfigParser()
+    cfg.read(os.path.expanduser('~/.nipaprc'))
+
+    # Parse command line arguments
+    from argparse import ArgumentParser
+    parser = ArgumentParser()
+    parser.add_argument('pattern', help="pattern to replace")
+    parser.add_argument('replacement', help="string to replace pattern with")
+    parser.add_argument('--username', help="username")
+    parser.add_argument('--password', help="password")
+    parser.add_argument('--host', help="NIPAP backend host")
+    parser.add_argument('--port', help="NIPAP backend port")
+    args = parser.parse_args()
+
+    # Set up NIPAP
+    auth_uri = "%s:%s@" % (args.username or cfg.get('global', 'username'),
+            args.password or cfg.get('global', 'password'))
+
+    xmlrpc_uri = "http://%(auth_uri)s%(host)s:%(port)s" % {
```

Comment:
guess this fails if I put in an IPv6 address here ;)

Most other scripts fail in a similar manner but perhaps we should start improving this. Can we always do [%(host)s]? I guess it works for IPv4 too, right?